frontend/rust-lib/flowy-database2/src/services/database_view/view_editor.rs

```diff
@@ -1101,11 +1101,23 @@ impl DatabaseViewEditor {
     });
   }
 
-  async fn gen_did_create_row_view_tasks(&self, row_id: RowId) {
+  async fn gen_did_create_row_view_tasks(&self, row: Row) {
+    let view_id = self.view_id.clone();
     let weak_sort_controller = Arc::downgrade(&self.sort_controller);
+    let weak_calculations_controller = Arc::downgrade(&self.calculations_controller);
     af_spawn(async move {
       if let Some(sort_controller) = weak_sort_controller.upgrade() {
-        sort_controller.read().await.did_create_row(row_id).await;
+        sort_controller
+          .read()
+          .await
+          .did_create_row(row.clone().id)
+          .await;
+      }
+
+      if let Some(calculations_controller) = weak_calculations_controller.upgrade() {
+        calculations_controller
+          .did_receive_row_changed(row, Some(view_id))
+          .await;
```

Comment:
You can pass the row as reference and only clone the row id in the sort controller